Submit Project for Evaluation

umair-rao opened the pull request:
**Hello and welcome to "SmartPocket"**

**In this project, I created a SmartPocket app that keeps track of all your expenses associated with categories.
It allows you to:**

create new transactions in order to keep track of your expenses
update/delete existing transactions
create new categories in order to classify your expenses by categories
see the calculated total amount of expenses per category
update/delete existing categories
General requirements:
there are no linter errors
correct Gitflow is used
work is documented in a professional way
Project requirements:
The project is deployed and accessible online.

I followed the design guidelines from [SnapScan](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=&) (Colors, Typography, Layout). I gave appropriate credit to the author in the Acknowledgements section of [README](https://github.com/umair-rao/Budget-app/blob/dev/README.md).

I created unit, request and integration tests for all the most important components of the app.

User input is always validated to make sure that anyone with bad intentions cannot compromise the app.

**Presentation**: https://www.loom.com/share/03984135f2774004a145dc15a651ff83?sid=3c1db79b-494d-495a-be63-f8013683add8

**Splash screen**
A simple page with the name of the app, and links to the sign up and log in pages (built with Devise).

**Sign up and login pages**
You are able to register in the app with your full name, email and password.
You can log into the app using email and password.
You can't access other pages unless you are logged in.

**Home page (categories page)**
When you log in, you are presented with the categories page.
For each category, you can see their name, icon and the total amount of all the transactions that belong to that category.
Whn you click on a category item, the application navigates to the transactions page for that category.
There is a button "add category" at the bottom that brings you to the page to create a new category.

**Transactions page**
For a given category, the list of transactions is presented, ordered by the most recent.
At the bottom of the page, you can see the total amount for the category.
There is a button "add transaction" at the bottom that brings you to the page to create a new transaction.
When you click on the "Back" button (<), you navigate to the home page.

**Additional features:**

if the category is empty (has no associated transactions), you will see a 'Delete' button that will delete this category
if you click on the category item, you will navigate to the page to edit this category
if you click on the transaction item, you will navigate to the Transaction Details page, where you can edit/delete this transaction
"Add a new category" page
You can fill out a form to create a new category, indicating their name and icon (both mandatory).
When you click the "Save" button to create the new category, you are taken to the home page on success.
When you click on the "Back" button (<), you navigate to the home page.

"Add a new transaction" page
You can fill out a form to create a new transaction with:

name (mandatory)
amount (mandatory)
categories (mandatory at least one)
When you click the "Save" button to create a new transaction, you are taken to the transactions page for that category.
When you click on the "Back" button (<), you navigate to the transactions page for that category.

umair-rao committed:
Submit Project for submission

OyePriscilla commented on README.md:
```diff
@@ -193,4 +193,4 @@ I am grateful to all previous project reviewers for their advice. Thanks to them
 
 This project is [MIT](./LICENSE) licensed.
 
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
- [ ] Please kindly ensure that when a category is created, users are redirected to the category homepage and not purchases index page. 👇 

<img width="358" alt="Untitled" src="https://github.com/umair-rao/Budget-app/assets/103386183/5e9bbbe8-bb74-4779-a187-073a2e06c381">

## Project Requirement 👇 

<img width="618" alt="Untitled" src="https://github.com/umair-rao/Budget-app/assets/103386183/8b35c3fe-02c0-4e22-b875-e8a0dbc2148f">

umair-rao replied:
Dear reviewer, It already navigates to transaction page you can all traction on the page (purchase) for which you have screenshot.

danmainah replied:
- [ ] @umair-rao Great work implementing the changes requested, however as the previous reviewer pointed out, when a new category is created, the user should be redirected to the categories page in your case `http://127.0.0.1:3000/groups` but in your app am getting redirected to `http://127.0.0.1:3000/groups/1/purchases`, this is part of the requirement, the reviewer might have used the wrong illustration very sorry for that but her point is valid, this can be changed in the `groups controller` where you can change the path to redirect page on line 14 from ` redirect_to group_purchases_path(@group)` to `redirect_to groups_path`.
![image](https://github.com/umair-rao/Budget-app/assets/61408860/a77df43b-6bf7-471b-8a24-8933aad40b21)

OyePriscilla commented on README.md:
```diff
@@ -193,4 +193,4 @@ I am grateful to all previous project reviewers for their advice. Thanks to them
 
 This project is [MIT](./LICENSE) licensed.
```
- [ ] Please kindly ensure that all commented codes are removed from the following files listed below 👇 

`/app/model/purchase.rb line 4`
`/app/spec/models/`all-the-files in your model's specs 👀 and
` /app/config/routes.rb line 12`.

umair-rao replied:
Dear reviewer, thank you for pointing out. I have removed all comments as mentioned by you.

OyePriscilla commented on README.md:
```diff
@@ -193,4 +193,4 @@ I am grateful to all previous project reviewers for their advice. Thanks to them
 This project is [MIT](./LICENSE) licensed.
 
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
```
- [ ] Kindly make a video presentation and add the link to the README file as stated in the requirement. ℹ️ 

<img width="780" alt="Untitled" src="https://github.com/umair-rao/Budget-app/assets/103386183/5e2cfd54-47b3-463e-92aa-3135bb069d14">

umair-rao replied:
Dear reviewer, as per requirement it is **not** compulsory to share only link in **Readme** file, However I have already provided link in **PR description**. I am also attaching link here for your reference. 

Presentation: https://www.loom.com/share/14eec9fdfc9f4b57983cbc2a480e586e?sid=da24f5ca-1efa-4914-b30c-e84fa3b4130d

danmainah replied:
- [ ] @umair-rao Great work on your presentation, however, I only saw your face at the beginning for a few seconds, It is a requirement to show the computer screen and your face simultaneously in a small picture during the presentation.
![image](https://github.com/umair-rao/Budget-app/assets/61408860/bee031e2-cf4c-4e39-91cb-3f15554d8bbe)
`NB Although it's not a requirement, it's nice having the presentation link in the readme, this makes it easier to access it by the public.`

OyePriscilla reviewed (changes requested):
Hi @umair-rao 👏 👏 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

![so-close](https://user-images.githubusercontent.com/103386183/196423863-0f522555-1b03-455e-a452-b822cdcfc398.gif)

## Highlights
Use ruby gems as software packages system.
Install Ruby on Rails framework.
Use controllers to handle requests and render empty views.
Use params from browser request in a safe way.
Use preprocessed html file with embedded Ruby code.
Use layouts and templates for shared content.
Use database migration files to maintain database schema.
Use validations for models.
Set up associations between models.
Build a webapp that requires the user to log in.
Use devise gem for authentication.

## Required Changes ♻️

- [ ] Please kindly look through the inline comments below and make necessary adjustments, I couldn't use the GitHub inline features correctly because I have less interaction with your PR (you've done all the project in the `dev` branch). ℹ️ 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

umair-rao committed:
Comments removed

umair-rao commented:
I have implemented changes as replied in the comments against the suggested requirements. Please refer to my comments reply. Thank you.

danmainah reviewed (changes requested):
Hi @umair-rao , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Implemented the requested change to remove all the commented code.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

umair-rao committed:
Update group controller

umair-rao commented:
Thank you for pointing this out with well explanation. I have updated the controller for group and also recorded video again showing my face. I have also added a video link into Readme. You are requested to please review.

Viedo Link: https://www.loom.com/share/03984135f2774004a145dc15a651ff83?sid=3c1db79b-494d-495a-be63-f8013683add8

umair-rao committed:
Update README.md

Munsa1 reviewed (approved):
Hi @umair-rao , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
